src/Differ/MapDiffer.php

```diff
@@ -176,6 +178,8 @@ private function isAssociative( array $array ) {
 	 * spot differences between array values. Very weird the native
 	 * function just ignores these...
 	 *
+	 * @SuppressWarnings(PHPMD.ShortVariable)
```

Comment:
I did not think about `$id`. You are right that we have way to many valid usecases for the rule to be enabled then. This applies to any rule - if we need to add ignore tags to many valid usecases, then we should not enable the rule.